moveit_core/kinematics_base/include/moveit/kinematics_base/kinematics_base.h

```diff
@@ -612,34 +615,17 @@ class KinematicsBase
   template <typename T>
   inline bool lookupParam(const std::string& param, T& val, const T& default_val) const
   {
-    ros::NodeHandle pnh("~");
-    if (pnh.hasParam(group_name_ + "/" + param))
-    {
-      val = pnh.param(group_name_ + "/" + param, default_val);
-      return true;
-    }
-
-    if (pnh.hasParam(param))
+    // TODO(henningkayser): reuse node and fix private namespace lookup from kinematics.yaml
+    auto node = rclcpp::Node::make_shared("kinematics_base.lookup_param");
+    std::vector<rclcpp::Parameter> param_results = node->get_parameters(
+        { group_name_ + "/" + param, param, "robot_description_kinematics/" + group_name_ + "/" + param,
+          "robot_description_kinematics/" + param });
+    if (!param_results.empty())
     {
-      val = pnh.param(param, default_val);
+      val = param_results.get_value<T>();
```

Comment:
This is a vector. I think you need to get the parameter at first index right? @henningkayser

Reply:
nit picky but I think this should be the order:
```cpp
 { 
    param,
    group_name_ + "/" + param,  
    "robot_description_kinematics/" + param,
    "robot_description_kinematics/" + group_name_ + "/" + param
}
```

Reply:
> This is a vector. I think you need to get the parameter at first index right? @henningkayser

@mlautman You're absolutely right. Fixed!

> nit picky but I think this should be the order:
> 
> ```c++
>  { 
>     param,
>     group_name_ + "/" + param,  
>     "robot_description_kinematics/" + param,
>     "robot_description_kinematics/" + group_name_ + "/" + param
> }
> ```

I kept the old order and don't think we should change it.
After all the specification (with group name) should override the default (without group name), no?

Reply:
fair enough. My nit was aesthetic